### PR TITLE
Add bubble sort

### PR DESCRIPTION
**Pull request by mbalamat**
I have added the BubbleSort algorithm.

### CODE REVIEW

File: bubblesort/BubbleSort.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+                    int temp = list[j];
+                    list[j] = list[j + 1];
+                    list[j + 1] = temp;
+                }
+            }
+        }
+
+        //Print Sorted Array
+        System.out.println("\n\nSorted Array is:");
+        for (int i = 0; i <= n - 1; ++i) {
```
Comment (gtklocker) [inline bubblesort/BubbleSort.java]: nit: Use < n instead of <= n - 1.

File: bubblesort/BubbleSort.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            System.out.print(" " + list[i]);
```
Comment (gtklocker) [inline bubblesort/BubbleSort.java]: This prints a space before the first element, we don't want that.
Also, we want to end with a newline.

File: bubblesort/BubbleSort.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import java.util.*;
+
+public class BubbleSort {
+    public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
+        int n = in.nextInt();
+        int list[] = new int[n];
+
+        System.out.println("Array before bubble sort:");
```
Comment (gtklocker) [inline bubblesort/BubbleSort.java]: Remove this message.

File: bubblesort/BubbleSort.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+                if (list[j] > list[j + 1]) {
+                    int temp = list[j];
+                    list[j] = list[j + 1];
+                    list[j + 1] = temp;
+                }
+            }
+        }
+
+        //Print Sorted Array
+        System.out.println("\n\nSorted Array is:");
```
Comment (gtklocker) [inline bubblesort/BubbleSort.java]: Remove this message.

File: bubblesort/BubbleSort.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+public class BubbleSort {
+    public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
+        int n = in.nextInt();
+        int list[] = new int[n];
+
+        System.out.println("Array before bubble sort:");
+        for (int i = 0; i <= n - 1; ++i) {
+            list[i] = in.nextInt();
+            System.out.print(" " + list[i]);
```
Comment (gtklocker) [inline bubblesort/BubbleSort.java]: Don't print the unsorted array.

File: bubblesort/BubbleSort.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        }
+
+        //BubbleSort
```
Comment (gtklocker) [inline bubblesort/BubbleSort.java]: nit: Remove this and all the other comments, they don't offer much.

File: bubblesort/BubbleSort.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        for (int i = 1; i < n; ++i) {
```
Comment (gtklocker) [inline bubblesort/BubbleSort.java]: Maybe you wanted to start with i = 0 here?

File: bubblesort/BubbleSort.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+            for (int j = 0; j < n - i; ++j) {
+                if (list[j] > list[j + 1]) {
+                    int temp = list[j];
+                    list[j] = list[j + 1];
+                    list[j + 1] = temp;
+                }
+            }
+        }
+        for (int i = 0; i <= n - 1; ++i) {
```
Comment (gtklocker) [inline bubblesort/BubbleSort.java]: nit: Use < n instead of <= n - 1.

File: bubblesort/BubbleSort.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+            System.out.print(list[i]);
```
Comment (gtklocker) [inline bubblesort/BubbleSort.java]: We need a space to separate each element. Output should be like this:

```
1 2 3 4 5
```

File: bubblesort/BubbleSort.java [hunk 10]
```diff
@@ -0,0 +1,25 @@
+import java.util.*;
+
+public class BubbleSort {
+    public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
+        int n = in.nextInt();
+        int list[] = new int[n];
+        for (int i = 0; i <= n - 1; ++i) {
+            list[i] = in.nextInt();
+        }
```
Comment (gtklocker) [inline bubblesort/BubbleSort.java]: nit: Separate the for blocks with newlines

File: bubblesort/BubbleSort.java [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+            for (int j = 0; j < n - i; ++j) {
+                if (list[j] > list[j + 1]) {
+                    int temp = list[j];
+                    list[j] = list[j + 1];
+                    list[j + 1] = temp;
+                }
+            }
+        }
+        for (int i = 0; i <= n - 1; ++i) {
+            System.out.print(list[i] + " ");
```
Comment (gtklocker) [inline bubblesort/BubbleSort.java]: This also adds a space after the last element, we don't want that!

File: bubblesort/BubbleSort.java [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+                }
+            }
+        }
+        String output = "";
+        output += list[0] + " ";
+        for (int i = 1; i <= n - 2; ++i) {
+            output += list[i] + " ";
+        }
+        output += list[n-1];
+        System.out.println(output);
```
Comment (gtklocker) [inline bubblesort/BubbleSort.java]: You are printing two newlines, we only need one. I'd rather you did this without using a variable to store the output.

File: bubblesort/BubbleSort.java [hunk 13]
```diff
@@ -0,0 +1,31 @@
+import java.util.*;
+
+public class BubbleSort {
+    public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
+        int n = in.nextInt();
+        int list[] = new int[n];
+        for (int i = 0; i <= n - 1; ++i) {
```
Comment (gtklocker) [inline bubblesort/BubbleSort.java]: Fix this.

**Comment by mbalamat** [comment]
@gtklocker rebase now?

**Comment by gtklocker** [comment]
Looks good. :+1: 

Yes, go ahead and squash those commits.

**Comment by gtklocker** [comment]
Good job. Here's to you. :beer:

**Pull request merged**